src/platform/packages/shared/kbn-es-query/src/project_routing/types.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+/**
+ * Project routing configuration for cross-project search (CPS).
+ *
+ * Used in serverless environments to control whether searches are scoped to a single project or span multiple projects.
+ *
+ * Examples:
+ * - undefined - Search across all projects (default)
+ * - '_alias:_origin' - Search only in the current project
+ *
+ * @public
+ */
+export type ProjectRouting = undefined | '_alias:_origin';
```

Comment:
Perhaps we should infer this type from the `projectRoutingSchema` using the `TypeOf` helper from `@kbn/config-schema`. We can safely import the `projectRoutingSchema` as a type from the server package.

Untested, please verify this works.
```suggestion
export type ProjectRouting = TypeOf<typeof projectRoutingSchema>`;
```